codemagic.yaml

```diff
@@ -0,0 +1,222 @@
+        adb emu kill
+    # Run app tests
+    - &stepAppRunTests
+      name: Test Android
+      script: |
+        cd project
+        ./gradlew connectedAndroidTest
+        result=$?
+        adb logcat -d > emulator.log
+        if [ $result -eq 0 ]
+        then
+          echo "Tests executed successfully"
+        else
+          echo "Tests failed"
+          exit 1
+        fi
+    # Run library sample app tests
+    - &stepLibAppRunTests
+      name: Test Android
+      script: |
+        cd projectlib
+        ./gradlew connectedAndroidTest
+        result=$?
+        adb logcat -d > emulator.log
+        if [ $result -eq 0 ]
+        then
+          echo "Tests executed successfully"
+        else
+          echo "Tests failed"
+          exit 1
+        fi
+
+# Flows
+workflows:
+  develop-app-builds:
+    name: Dev app builds
+    instance_type: linux_x2
+    environment:
+      java: 17
+      groups:
+        - keystore_credentials
+    triggering:
+      events:
+        - push
+        - pull_request
+      branch_patterns:
+        - pattern: 'develop'
+          include: true
+          source: true
+    scripts:
+      - *stepAppLocalSetup
+      - *stepAppCheckCode
+      - *stepStartEmulator
+      - *stepAppRunTests
+      - *stepStopEmulator
+      - *stepAppBuildDebug
+    artifacts: *appArtifacts
+    publishing:
+      email:
+        recipients: *debugEmails
+  develop-lib-builds:
+    name: Dev lib builds
+    instance_type: linux_x2
+    environment:
+      java: 17
+    triggering:
+      events:
+        - push
+        - pull_request
+      branch_patterns:
+        - pattern: 'develop'
+          include: true
+          source: true
+    scripts:
+      - *stepLibLocalSetup
+      - *stepLibCheckCode
+      - *stepStartEmulator
+      - *stepLibAppRunTests
+      - *stepStopEmulator
+      - *stepLibBuild
+    artifacts: *libArtifacts
+    publishing:
+      email:
+        recipients: *debugEmails
+  merge-requests-app:
+    name: Merge requests app
+    instance_type: linux_x2
+    environment:
+      java: 17
+      groups:
+        - keystore_credentials
+    triggering:
+      events:
+        - pull_request
+      branch_patterns:
+        - pattern: 'develop'
+          include: true
+          source: false
+      cancel_previous_builds: true
+    scripts:
+      - *stepAppLocalSetup
+      - *stepAppCheckCode
+      - *stepStartEmulator
+      - *stepAppRunTests
+      - *stepStopEmulator
+      - *stepAppBuildDebug
+    publishing:
+      email:
+        recipients: *debugEmails
+        notify:
+          success: false
+  merge-requests-lib:
+    name: Merge requests lib
+    instance_type: linux_x2
+    environment:
+      java: 17
+    triggering:
+      events:
+        - pull_request
+      branch_patterns:
+        - pattern: 'develop'
+          include: true
+          source: false
+      cancel_previous_builds: true
+    scripts:
+      - *stepLibLocalSetup
+      - *stepLibCheckCode
```

Comment:
I understand that the lib only has sample tests? Jus that if "develop-lib-builds" runs tests also then it would probably be safer to also run these for PRs. Or if we do not care about the tests because they are just samples then maybe not run them at either.

Reply:
Added tests to PR workflow as well.